Gems/AtomLyIntegration/CommonFeatures/Code/Source/Mesh/MeshComponentController.cpp

```diff
@@ -699,6 +701,74 @@ namespace AZ
             return Aabb::CreateNull();
         }
 
+        void MeshComponentController::GetVisibleGeometry(AzFramework::VisibleGeometryContainer& geometryContainer) const
+        {
+            // Attempt to copy the triangle list geometry data out of the model asset into the visible geometry structure
+            const auto& modelAsset = GetModelAsset();
+            if (!modelAsset.IsReady() || modelAsset->GetLodAssets().empty())
+            {
+                return;
+            }
+
+            // This will only extract data from the first LOD. It might be necessary to make the LOD selectable.
+            const auto& lodAsset = modelAsset->GetLodAssets().front();
+            if (!lodAsset.IsReady())
+            {
+                return;
+            }
+
+            const AZ::Name positionName{ "POSITION" };
+            for (const auto& mesh : lodAsset->GetMeshes())
+            {
+                // Get the index buffer data, confirming that the asset is valid and indices are 32 bit integers. Other formats are
+                // currently not supported.
+                const AZ::RPI::BufferAssetView& indexBufferView = mesh.GetIndexBufferAssetView();
+                const AZ::RHI::BufferViewDescriptor& indexBufferViewDesc = indexBufferView.GetBufferViewDescriptor();
+                const AZ::Data::Asset<AZ::RPI::BufferAsset> indexBufferAsset = indexBufferView.GetBufferAsset();
+                if (!indexBufferAsset.IsReady() || indexBufferViewDesc.m_elementSize != sizeof(uint32_t))
+                {
+                    continue;
```

Comment:
Feels like all of these "wrong format" conditionals should print out a warning maybe instead of silently getting dropped?

Reply:
Or better yet return an AZ::Outcome.

Reply:
Fair point, though I'm personally not bought into championing the Outcome/expected path as a style choice. The warnings still need to get printed out _somewhere_, and the further up the chain they get printed, the harder it is to trace back to the root cause. I've had that problem with the JSON serialization system. If the theory is that something up the chain could potentially _handle_ the error, then the returned errors would ideally be an error code, not a string...

Reply:
I am good with returning an error code instead of a string.
However logging messages via `AZ_Error` or `AZ_Warning` in the function itself causes a spam of error messages that has spiraled out of control in the engine and we are looking to address.

There is big effort to simplify the Error message itself with several tickets created by the UI/UX SIG grouup
https://github.com/o3de/o3de/issues/11666
https://github.com/o3de/o3de/issues/9537


Still further, the O3DE logging system can force failures on to callers of an API, when they don't desire to fail on a particular result.
Logging an `AZ_Error` causes a running GTest test case to fail, even though the test itself might not care about the internal error message call.
That is main reason we have the [AZ_START_TRACE_SUPPRESSION/AZ_STOP_TRACE_SUPPRESSION](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/UnitTest/UnitTest.h#L297-L299) macros, because there is frequently times when a function instead of returning an error code, logs the error message itself, requiring the test to workaround calls to that function. See this MathUtils test [here](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/Tests/Math/MathUtilsTests.cpp#L167-L176).

I have actually been using in my code less AZ::Outcome/AZStd::expected and more of a "result" structure that contains an error code and a message string.

For example the `SettingsRegistry::MergeSettings*` functions returns a result structure with an enum error code and a message string.
https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Settings/SettingsRegistry.h#L382-L418

Reply:
Alternatively with the new Archive system, it uses a more detailed result structure that stores information about the operation as well as provides an `operator bool` to convert the result to a boolean context
https://github.com/o3de/o3de/blob/development/Gems/Archive/Code/Include/Archive/Clients/ArchiveReaderAPI.h#L114-L145
https://github.com/o3de/o3de/blob/development/Gems/Archive/Code/Include/Archive/Tools/ArchiveWriterAPI.h#L123-L149

Reply:
Great suggestions. However, outcome does not align with many of the other buses or choices made for most of these components. Outcome also requires additional handling up the call stack that deviates from the rest of the handling. issues with logging and spam sound like a much larger problem that needs to be addressed globally, beyond the scope of this change. Somewhat related, I stripped outcome out of systems that I've previously worked on in favor of placing an object on the stack just before some operation, having it record trace messages, errors, etc. Then presenting them to the user if there were any errors or warnings.